hardware.md

```diff
@@ -99,33 +99,40 @@ The board has a physical jumper to switch 3.3V power supply to the target.
+## ST-Link
 
-For a generic STM32F103 board, look if the swlink platform can handle it. Swlink reuses the SWD pins.
-You can also follow [these](https://medium.com/@paramaggarwal/converting-an-stm32f103-board-to-a-black-magic-probe-c013cf2cc38c#.btn6lnwqe) instructions for a solution to not reuse the SWD pins.
+You can compile the firmware for an ST-Link debugger. These are often included on ST Discovery and Nucleo evaluation boards. You can replace [the ST-Link firmware on your discovery board](http://esden.net/2014/12/29/black-magic-discovery/) with Black Magic Probe Firmware. Usually this leads to more stable operation and easier use than the ST tools. There are several tutorials how to do it, even if you don't have a JTAG adapter yet and no way to bootstrap. :)
 
-## ST Link V2
+## ST-Link V2
 
-Restoring StlinkV2 BMP support with recent original ST firmware is a work in progress at the moment, according to the [issue #504](https://github.com/blacksphere/blackmagic/issues/504#issuecomment-525399540).
+Instructions to restore an ST-Link V2 with recent original ST firmware are found on [GitHub stlink](https://github.com/blackmagic-debug/blackmagic/tree/main/src/platforms/stlink#reverting-to-original-st-firmware-with-running-bmp-firmware).
 
 ## F4 Discovery
 
-You can run the Black Magic Probe firmware on the "target" processor of an ST F4 discovery board. This among others useful if you want to bootstrap your F4 Discovery board ST Link programmer without having any other means of programming it.
+You can run the Black Magic Probe firmware on the "target" processor of an ST F4 discovery board. This is useful if you want to bootstrap your F4 Discovery board ST-Link programmer without having any other means of programming it.
 
 ## SW Link
 
 This target is the programmer integrated on the STM8S Discovery board.
 
 ## Bluepill/Blackpill V1 minimal development boards
 
-The swlink platform can also be used for development boards with header accessible SWD pins. Swlink firmware remaps SWD debug pins as bit-bang SWD output. Full JTAG and more infrastructure is accessible on the header rows.
+The swlink platform can also be used for development boards with header accessible SWD pins. Swlink firmware remaps the SWD debug pins as bit-bang SWD output. Full JTAG and more infrastructure is accessible on the header rows.
 
-## Launchpad ICDI
+You can also follow [these](https://medium.com/@paramaggarwal/converting-an-stm32f103-board-to-a-black-magic-probe-c013cf2cc38c#.btn6lnwqe) instructions for a solution to not reuse the SWD debug pins.
```

Comment:
Just taken a look at this guide and it's pointing at the old repository URL and provides incorrect build instructions. Need to either write or find more accurate instructions.

Reply:
I have removed the link (as it provides incorrect build instructions) and updated the section on the hardware page for Blue Pill boards. This should clarify which platform to use, to not reuse the SWD debug pins.